### ISSUE TITLE
Too much noise, too many pods

### ISSUE THREAD

**Issue by migmartri** [kubeapps/kubeapps#668]
My namespace is polluted with many instances of the sync jobs. It seems that the list grows on every chart upgrade. It would be great if we could limit the number of pods kept for reference.
```
repo-sync-bitnami-tbr7j-6bvkn                                  0/1       Completed   0          21h
apprepo-sync-bitnami-tt4mj-tc9fl                                  0/1       Completed   0          8d
apprepo-sync-bitnami-vv2nl-kdq9f                                  0/1       Completed   0          1d
apprepo-sync-bitnami-z22l7-wj2xq                                  0/1       Completed   0          1d
apprepo-sync-incubator-1537552800-flk79                           0/1       Completed   0          2h
apprepo-sync-incubator-1537556400-kf5pf                           0/1       Completed   0          1h
apprepo-sync-incubator-1537560000-4jtz4                           0/1       Completed   0          29m
apprepo-sync-incubator-2tz4j-jfwrv                                0/1       Completed   0          1h
apprepo-sync-incubator-2vch8-wczsj                                0/1       Completed   0          3m
apprepo-sync-incubator-6b2d9-kqqxq                                0/1       Completed   0          21h
apprepo-sync-incubator-84lwv-9m29p                                0/1       Completed   0          22h
apprepo-sync-incubator-bm5bt-vn976                                0/1       Completed   0          1m
apprepo-sync-incubator-hgxds-54hwx                                0/1       Completed   0          8d
apprepo-sync-incubator-kzhwf-h624b                                0/1       Completed   0          8d
apprepo-sync-incubator-lk7fm-mg9x6                                0/1       Completed   0          3d
apprepo-sync-incubator-qh7nj-q8ssj                                0/1       Completed   0          21h
apprepo-sync-incubator-qj6hh-nbwgw                                0/1       Completed   0          1d
apprepo-sync-incubator-smvcv-h4zhh                                0/1       Completed   0          1d
apprepo-sync-incubator-ts5hv-zt2tr                                0/1       Completed   0          1h
apprepo-sync-incubator-xkzgn-db8fb                                0/1       Completed   0          5m
apprepo-sync-stable-1537552800-qrzhz                              0/1       Completed   0          2h
apprepo-sync-stable-1537556400-2tdkv                              0/1       Completed   0          1h
apprepo-sync-stable-1537560000-m5zsf                              0/1       Completed   0          29m
apprepo-sync-stable-525wg-l4lt5                                   0/1       Completed   0          8d
apprepo-sync-stable-bwgdp-wzhbp                                   0/1       Completed   0          3d
apprepo-sync-stable-d2756-8wwks                                   0/1       Completed   0          1h
apprepo-sync-stable-d7txl-8dj8b                                   0/1       Completed   0          5m
apprepo-sync-stable-dfsv4-cmb9t                                   0/1       Completed   0          8d
apprepo-sync-stable-gnkdr-wmklb                                   0/1       Completed   0          8d
apprepo-sync-stable-js5qd-c27vz                                   0/1       Completed   0          1m
apprepo-sync-stable-lbm6g-q26jb                                   0/1       Completed   0          8d
apprepo-sync-stable-lhw9w-b85np                                   0/1       Completed   0          21h
apprepo-sync-stable-mc4ld-kr829                                   0/1       Completed   0          3m
apprepo-sync-stable-n2nvn-krxqw                                   0/1       Completed   0          21h
apprepo-sync-stable-qmgxd-pn7sk                                   0/1       Completed   0          22h
apprepo-sync-stable-w49vh-5frc8                                   0/1       Completed   0          1d
apprepo-sync-stable-xbvzz-4p97g                                   0/1       Completed   0          1h
apprepo-sync-stable-zbgmm-dl8fx                                   0/1       Completed   0          1d
apprepo-sync-svc-cat-1537552800-d7t6k                             0/1       Completed   0          2h
apprepo-sync-svc-cat-1537556400-szwqp                             0/1       Completed   0          1h
apprepo-sync-svc-cat-1537560000-wm67l                             0/1       Completed   0          29m
apprepo-sync-svc-cat-65qmp-76nrk                                  0/1       Completed   0          1h
apprepo-sync-svc-cat-f97nd-6whgt                                  0/1       Completed   0          1m
apprepo-sync-svc-cat-fx77b-hxpmb                                  0/1       Completed   0          1d
apprepo-sync-svc-cat-h2gxd-vkzg8                                  0/1       Completed   0          3m
apprepo-sync-svc-cat-hgd9f-t58q5                                  0/1       Completed   0          21h
apprepo-sync-svc-cat-jd4zk-bqw5h                                  0/1       Completed   0          21h
apprepo-sync-svc-cat-jksxs-x62g7                                  0/1       Completed   0          22h
apprepo-sync-svc-cat-jmx89-nhftp                                  0/1       Completed   0          8d
apprepo-sync-svc-cat-klftx-2zznd                                  0/1       Completed   0          5m
apprepo-sync-svc-cat-ptbvh-nxsnh                                  0/1       Completed   0          1d
apprepo-sync-svc-cat-r6fhv-vwwql                                  0/1       Completed   0          1h
apprepo-sync-svc-cat-szbpb-5qrw5                                  0/1       Completed   0          8d
apprepo-sync-svc-cat-tkf8q-gl8tx                                  0/1       Completed   0          3d
```

**Comment by prydonius** [comment]
Yeah, I believe each time apprepository-controller gets restarted it'll create a new manual sync job, also if you click refresh in the UI. These aren't cleaned up. One option would be to add something like this: https://github.com/dtan4/k8s-job-cleaner to our deployment as a CronJob.

**Comment by andresmgot** [comment]
at some point this problem will go away: https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/#ttl-mechanism-for-finished-jobs

**Comment by prydonius** [comment]
Indeed, let's see if this moves to beta in 1.13, we'd still probably want to wait for 1.14/1.15 before we adopt it.

**Comment by mahendrabagul** [comment]
> Yeah, I believe each time apprepository-controller gets restarted it'll create a new manual sync job, also if you click refresh in the UI. These aren't cleaned up. One option would be to add something like this: https://github.com/dtan4/k8s-job-cleaner to our deployment as a CronJob.

You can use this code for time being. https://github.com/mahendrabagul/k8s-job-reaper It creates another cronjob to delete the jobs in specified namespace.

I have raised PR to original project but its yet to get merged.

**Comment by andresmgot** [comment]
This should be fixed after #2297
